docs/src/main/tut/chisel3/chisel3-vs-compat.md

```diff
@@ -0,0 +1,148 @@
+---
+layout: docs
+title:  "chisel3 vs. compatibility mode"
+section: "chisel3"
+---
+
+Chisel 3 supports "compatibility mode" which serves the purpose of bridging
+the semantic gap between Chisel 2 and Chisel 3 with Chisel2-like semantics.
+Chisel code using this mode depends on Chisel 3, yet the Scala code has
+`import Chisel._` instead of the normal `import chisel3._`. This page serves
+to document the differences for helping migrate from compatibility mode to the
+safer and more modern Chisel 3 semantics.
+
+### Connections
+
+#### Compatibility mode
+
+In compatibility mode, both connection operators `:=` and `<>` are actually the same.
+They both are bidirectional and noncommutative--the right side will be treated as
+the source and the left side will be treated as the sink.
+
+```scala
+import Chisel._
+class MyModule extends Module {
+  val io = new Bundle {
+    val in = Decoupled(UInt(width = 4)).flip
+    val out = Decoupled(UInt(width = 4))
+  }
+  io.in <> io.out // This works
+  io.in := io.out // This is equivalent
+  io.out <> io.in // This will error
+}
+```
+Furthermore, error detection and reporting is defered to FIRRTL compilation.
+
+#### Chisel 3
+
+In Chisel 3, `:=` is refered to as "monoconnect" and `<>` is called "bulkconnect".
+
+* Monoconnect
+`:=` treats everything on the left-hand side as a sink, even if the type is
+bidirectional. This means it cannot be used to drive bidirectional output ports,
+but can be used to drive wires from bidirectional inputs or outputs to "monitor"
+the the full aggregate, ignoring directions.
+
+* Bulkconnect
```

Comment:
I think this is actually `BiConnect`, yes? Or should this be referred to as Bulk Connect?

Reply:
Traditionally people called it "bulk connect" but I agree "biconnect" is better anyway